--- wpa_supplicant-1.0/wpa_supplicant/Makefile.orig	2012-05-09 23:56:09.000000000 +0200
+++ wpa_supplicant-1.0/wpa_supplicant/Makefile	2012-05-22 17:15:38.836468705 +0200
@@ -53,6 +53,8 @@
 	install -D $(<) $(@)
 
 install: $(addprefix $(DESTDIR)$(BINDIR)/,$(BINALL))
+	mkdir -p $(DESTDIR)/etc
+	cp wpa_supplicant.conf $(DESTDIR)/etc
 	$(MAKE) -C ../src install
 
 OBJS = config.o
@@ -1483,6 +1483,8 @@
 wpa_gui-qt4/lang/wpa_gui_de.qm: wpa_gui-qt4/lang/wpa_gui_de.ts
 	lrelease-qt4 wpa_gui-qt4/wpa_gui.pro
 
+.PHONY: wpa_gui-qt4
+
 wpa_gui-qt4: wpa_gui-qt4/Makefile wpa_gui-qt4/lang/wpa_gui_de.qm
 	$(MAKE) -C wpa_gui-qt4
 
