--- wpa_supplicant/Makefile.orig	2010-09-08 15:34:15.413763615 +0100
+++ wpa_supplicant/Makefile	2010-09-08 15:36:34.555511910 +0100
@@ -1357,7 +1357,7 @@
 	qmake -o wpa_gui-qt4/Makefile wpa_gui-qt4/wpa_gui.pro 
 
 wpa_gui-qt4/lang/wpa_gui_de.qm: wpa_gui-qt4/lang/wpa_gui_de.ts
-	lrelease wpa_gui-qt4/wpa_gui.pro
+	lrelease-qt5 wpa_gui-qt4/wpa_gui.pro
 
 wpa_gui-qt4: wpa_gui-qt4/Makefile wpa_gui-qt4/lang/wpa_gui_de.qm
 	$(MAKE) -C wpa_gui-qt4
