diff -urNp -x '*.orig' wpa_supplicant-2.9.org/wpa_supplicant/Makefile wpa_supplicant-2.9/wpa_supplicant/Makefile
--- wpa_supplicant-2.9.org/wpa_supplicant/Makefile	2021-09-26 23:01:54.976837708 +0200
+++ wpa_supplicant-2.9/wpa_supplicant/Makefile	2021-09-26 23:01:55.070171269 +0200
@@ -3,7 +3,7 @@ CC=gcc
 endif
 
 ifndef CFLAGS
-CFLAGS = -MMD -O2 -Wall -g
+CFLAGS = -MMD $(OPTCFLAGS) -Wall -g
 endif
 
 ifdef LIBS
