diff -Nur wpa_supplicant-0.5.4/wpa_supplicant/Makefile wpa_supplicant-0.5.4.OPTCFLAGS/wpa_supplicant/Makefile
--- wpa_supplicant-0.5.4/wpa_supplicant/Makefile	2006-08-03 19:30:41.000000000 +0000
+++ wpa_supplicant-0.5.4.OPTCFLAGS/wpa_supplicant/Makefile	2006-08-03 19:30:42.000000000 +0000
@@ -3,7 +3,7 @@
 endif
 
 ifndef CFLAGS
-CFLAGS = -MMD -O2 -Wall -g
+CFLAGS = -MMD $(OPTCFLAGS) -Wall -g
 endif
 
 # Include directories for CVS version
