--- wpa_supplicant-0.3.8.orig/wpa_supplicant/Makefile	2005-02-14 02:55:47.000000000 +0100
+++ wpa_supplicant-0.3.8/wpa_supplicant/Makefile	2005-04-07 21:55:05.803706440 +0200
@@ -38,6 +38,8 @@
 	echo CONFIG_WIRELESS_EXTENSION=y >> .config
 
 install: all
+	mkdir -p $(DESTDIR)/etc
+	cp wpa_supplicant.conf $(DESTDIR)/etc
 	mkdir -p $(DESTDIR)$(BINDIR)
 	for i in $(ALL); do cp $$i $(DESTDIR)$(BINDIR)/$$i; done
 	$(MAKE) -C ../src install
