diff -Nur wpa_supplicant-0.5.4/Makefile wpa_supplicant-0.5.4.kill-dynamic_eap_methods/Makefile
--- wpa_supplicant-0.5.4/Makefile	2006-08-03 19:31:14.000000000 +0000
+++ wpa_supplicant-0.5.4.kill-dynamic_eap_methods/Makefile	2006-08-03 19:31:15.000000000 +0000
@@ -9,7 +9,7 @@
 # Include directories for CVS version
 CFLAGS += -I. -I../utils -I../hostapd
 
-ALL=wpa_supplicant wpa_passphrase wpa_cli dynamic_eap_methods
+ALL=wpa_supplicant wpa_passphrase wpa_cli
 
 all: verify_config $(ALL)
 
